src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -73,7 +87,11 @@ class CoreModuleImpl @Inject() (
     marathonSchedulerDriverHolder,
 
     // internal core dependencies
-    offerMatcherManagerModule.globalOfferMatcher)
+    StopOnFirstMatchingOfferMatcher(
+      offerMatcherReconcilerModule.offerMatcherReconciler,
+      offerMatcherManagerModule.globalOfferMatcher
```

Comment:
I understand that you want to stop on the first match for the reconciler.
Not 100% sure, why the same logic applies to the global offer matcher?